### PR TITLE
Feature-7580 dynamic queue small exchange set

### PR DESCRIPTION
**Pull request by Siddartha-tech**
Code changes
Unit test

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/appsettings.json [hunk 1]
```diff
@@ -65,8 +65,9 @@
     "LargeExchangeSetAccountKey": "",
     "LargeExchangeSetInstance": 1,
     "StorageContainerName": "ess-fulfilment",
-    "QueueName": "ess-1-fulfilment",
-    "LargeExchangeSetSizeInMB": 300
+    "DynamicQueueName": "ess-{0}-fulfilment",
```
Comment (janvi13603) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/appsettings.json:68]: As we have done changes for dynamic queue and storage based on exchange set size, can we remove below from appsettings.json
"StorageAccountName": "",
    "StorageAccountKey": "", 
Also make necessary changes in appsettings.local.overrrides.json to keep both settings in sync

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/AzureBlobStorageService.cs [hunk 2]
```diff
@@ -100,6 +114,39 @@ public async Task<SalesCatalogueProductResponse> DownloadSalesCatalogueResponse(
...
+            {
+                return (mediumExchangeSetInstance.GetInstanceCount(storageConfig.Value.MediumExchangeSetInstance), ExchangeSetType.MediumExchangeSet.ToString());
+            }
+            else if (fileSizeInMB > storageConfig.Value.LargeExchangeSetSizeInMB)
+            {
+                return (largeExchangeSetInstance.GetInstanceCount(storageConfig.Value.LargeExchangeSetInstance), ExchangeSetType.LargeExchangeSetInstance.ToString());
+            }
+            else
+            {
+                return (smallExchangeSetInstance.GetInstanceCount(storageConfig.Value.MediumExchangeSetInstance), ExchangeSetType.SmallExchangeSet.ToString());
```
Comment (janvi13603) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/AzureBlobStorageService.cs:132]: Kindly change (storageConfig.Value.MediumExchangeSetInstance) to (storageConfig.Value.SmallExchangeSetInstance)

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/appsettings.json (hunk 1 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/appsettings.json:66]: Yes removed unused properties

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/AzureBlobStorageService.cs (hunk 2 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/AzureBlobStorageService.cs]: Yes updated accordingly.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Configuration/ILargeExchangeSetInstance.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿namespace UKHO.ExchangeSetService.Common.Configuration
+{
+    public interface ILargeExchangeSetInstance
+    {
+        int GetCurrentInstaceCount();
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Configuration/ILargeExchangeSetInstance.cs]: Yes made the changes all the respective files

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/AzureMessageQueueHelper.cs [hunk 4]
```diff
@@ -1,24 +1,35 @@
...
+            var queue = string.Format(essFulfilmentStorageconfig.Value.DynamicQueueName, instanceCount);
             // Create the queue client.
-            QueueClient queueClient = new QueueClient(storageAccountConnectionString, essFulfilmentStorageConfiguration.QueueName);
+            QueueClient queueClient = new QueueClient(storageAccountConnectionString, queue);
 
             // convert message to base64string          
             var messageBase64String = System.Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes(message));
             // Send a message to the queue
             await queueClient.SendMessageAsync(messageBase64String);
+            logger.LogInformation(EventIds.AddedMessageInQueueSCSResponseStored.ToEventId(), "Added message in Queue:{queue} with Sales catalogue response uri for BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", queue, batchId, correlationId);
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/AzureMessageQueueHelper.cs]: Updated EventId accordingly

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Models/Enums/ExchangeSetType.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿namespace UKHO.ExchangeSetService.Common.Models.Enums
+{
+    public enum ExchangeSetType
+    {
+        SmallExchangeSet = 1,
+        MediumExchangeSet = 2,
+        LargeExchangeSetInstance = 3
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Models/Enums/ExchangeSetType.cs]: Renamed enum

**Review by janvi13603 — APPROVED** [review]
Thanks for fixing review comments.

**Review by vijendr13545 — APPROVED** [review]
LGTM

**Review by martyn-fewtrell — APPROVED** [review]
Thanks for the updates

**Pull request merged**